src/index.ts

```diff
@@ -13,10 +13,15 @@ import { Options as RemarkRehypeOptions } from 'mdast-util-to-hast';
 import remarkToRehype from 'remark-rehype';
 import rehypeReact, { Options as RehypeReactOptions } from 'rehype-react';
 
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
+
 export interface UseRemarkOptions {
   remarkParseOptions?: Partial<RemarkParseOptions>;
   remarkToRehypeOptions?: RemarkRehypeOptions;
-  rehypeReactOptions?: Partial<RehypeReactOptions<typeof createElement>>;
+  rehypeReactOptions?: PartialBy<
+    RehypeReactOptions<typeof createElement>,
+    'createElement'
+  >;
```

Comment:
with the smarter with/without node types in rehype react, this needed an update to only make `createElement` optional, and keep the rest of the types unchanged.